middleware/faq.rst

```diff
@@ -382,14 +382,17 @@ When that graph is empty, it means it thinks there's no work to be done, and unf
 
 The `QuantumGraph` generation algorithm begins with a large SQL query (a complicated invocation of `Registry.queryDataIds`, actually), where the result rows are essentially data IDs and the result columns are all of the dimensions referenced by any task or dataset type in the pipeline.
 Queries for all `"regular input" <connectionTypes.Input>` datasets (i.e. not `PrerequisiteInputs <connectionTypes.PrerequisiteInput>`") are included as subqueries, spatial and temporal joins are automatically included, and the user-provided query expression is translated into an equivalent SQL ``WHERE`` clause.
-That means there are many ways to get no result rows - and hence an empty graph - without much information about what was missing.
-Some common possibilities include:
+That means there are many ways to get no result rows - and hence an empty graph.
 
-- There are no instances of an input dataset type in the input collections.
-- There are no dimension records of a needed type.
-- There is no spatial or temporal overlap between existing datasets and the data IDs accepted by the query expression (e.g. the ``visits`` don't overlap the ``patches``).
+Sometimes we can tell what will go wrong even before the query is executed - the butler maintains a summary of which dataset types are present each each collection, so if the input collections don't have any datasets of a needed type at all, a warning log message will be generated stating the problem.
+This will also catch most cases where a pipeline is misconfigured such that what should be an intermediate dataset isn't actually being produced in the pipeline, because it will appear instead as an overall-input that (usually) won't be present in those input collections.
```

Comment:
I don't understand the significance of the hypen in "overall-input".

Reply:
I mean "an input to a task that is not produced by another task in the pipeline".  If you'd find it less confusing to drop the hyphen, I'll do that.

Reply:
Just fixing it anyway; hyphen doesn't make sense there grammatically regardless, and I think that's what threw you off.